src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,140 @@
+package nearsoft.academy.bigdata.recommendation;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import org.apache.mahout.cf.taste.common.TasteException;
+import org.apache.mahout.cf.taste.impl.model.file.FileDataModel;
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.Recommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+import java.io.*;
+import java.util.*;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Created by marco on 20/09/16.
+ */
+public class MovieRecommender {
+    private static final String OUTPUT_FILEPATH = "movies.csv";
+    private static final String PRODUCT_KEY = "product/productId: ";
+    private static final String USER_KEY = "review/userId: ";
+    private static final String SCORE_KEY = "review/score: ";
+
+    private int totalReviews;
+    private HashMap<String, Integer> users;
+    private HashBiMap<String, Integer> products;
+    private Recommender recommender;
+
+
+    public MovieRecommender(String filePath) {
+        this.users = new HashMap<String, Integer>();
+        this.products = HashBiMap.create();
+        File csvFile = generateCSV(filePath);
+        buildRecommender(csvFile);
+    }
+
+    private File generateCSV(String filePath) {
```

Comment:
Creating a map like would make the code harder to read. You should create an intermediate class that contains both properties you specify there. That way you can access to that model and see both properties. Remember to encapsulate correctly the properties (create setter and getter properly)